lte/gateway/c/oai/tasks/sgw/pgw_config.c

```diff
@@ -105,6 +107,34 @@ int pgw_config_process(pgw_config_t* config_pP) {
         LOG_SPGW_APP, "Found SGI interface MTU=%d\n", config_pP->ipv4.mtu_SGI);
     close(fd);
   }
+#if (!SGW_ENABLE_SESSIOND_AND_MOBILITYD)
+  struct in_addr each_ip_addr;
+  for (int i = 0; i < config_pP->num_ue_pool; i++) {
+    uint32_t range_low_hbo  = ntohl(config_pP->ue_pool_range_low[i].s_addr);
+    uint32_t range_high_hbo = ntohl(config_pP->ue_pool_range_high[i].s_addr);
+    uint32_t tmp_hbo        = range_low_hbo ^ range_high_hbo;
+    uint8_t nbits           = 32;
+    while (tmp_hbo) {
+      tmp_hbo = tmp_hbo >> 1;
+      nbits -= 1;
+    }
+    uint32_t network_hbo = range_high_hbo & (UINT32_MAX << (32 - nbits));
+    uint32_t netmask_hbo = 0xFFFFFFFF << (32 - nbits);
+    config_pP->ue_pool_network[i].s_addr = htonl(network_hbo);
+    config_pP->ue_pool_netmask[i].s_addr = htonl(netmask_hbo);
+
+    // Any math should be applied onto host byte order i.e. ntohl()
+    each_ip_addr.s_addr = config_pP->ue_pool_range_low[i].s_addr;
+    while (htonl(each_ip_addr.s_addr) <=
+           htonl(config_pP->ue_pool_range_high[i].s_addr)) {
```

Comment:
isn't this already in network byte order?

Reply:
same with `each_ip_addr.s_addr`, it is copied from `config_pP->ue_pool_range_low[i].s_addr` and looks like it is also network byte order.

Reply:
Please double check whether these should be instead converted into host byte order instead as we are doing integer ranking and comparisons.

Reply:
In fact why don't we simply iterate with `for (int i = range_low_hbo; i <= range_high_hbo; ++i) {  ip4_ref              = calloc(1, sizeof(conf_ipv4_list_elm_t)); ip4_ref->addr.s_addr = htonl(i); STAILQ_INSERT_TAIL(&config_pP->ipv4_pool_list, ip4_ref, ipv4_entries);}`